index.js

```diff
@@ -167,7 +167,8 @@ function setupWebSocketProxy (fastify, options, rewritePrefix) {
     wsProxy.findUpstream = function (request) {
       const source = new URL(request.url, 'ws://127.0.0.1')
       const upstream = options.replyOptions.getUpstream(request, '')
-      const target = new URL(source.pathname, upstream)
+      const isIpc = upstream.indexOf('ws+unix:') === 0 || upstream.indexOf('wss+unix:') === 0
+      const target = isIpc ? new URL(upstream) : new URL(source.pathname, upstream)
```

Comment:
A bit late for the reply. But the implementation here is incomplete.
The proper pathname should be `sock:path`.

Which means you should append the socket path before the `source.pathname` instead of only provide the socket.
```js
      const source = new URL(request.url, 'ws://127.0.0.1')
      const upstream = options.replyOptions.getUpstream(request, '')
      const isIpc = upstream.indexOf('ws+unix:') === 0 || upstream.indexOf('wss+unix:') === 0
      const target = new URL(source.pathname, upstream)
      target.protocol = upstream.indexOf('http:') === 0 ? 'ws:' : 'wss'
      target.search = source.search

      if(isIpc) {
        const { pathname } = new URL(upstream)
        target.pathname = pathname.indexOf(':') === -1 ? pathname + ':' + target.pathname : pathname + target.pathname
      }
 ```

Reply:
This fix leads to several issues. Firstly, the prefix handling is broken, though prefix (IMHO) is more useful in an ipc workflow than additional path support. Secondly, should it call yet one new Url(...) when source path is already parsed in `const source = ...`?  Furthermore, ipc socket related code is splitted into two blocks. Code flow isn't so clean.  My proposal is consider the: additional path support, valid prefix handling and straightforward code flow:
```js
      const source = new URL(request.url, 'ws://127.0.0.1')
      const upstream = options.replyOptions.getUpstream(request, '')
      const isIpc = upstream.indexOf('ws+unix:') === 0 || upstream.indexOf('wss+unix:') === 0
      let target;
        if(isIpc){
        target = new URL(upstream);
        const path = source.pathname.startsWith(fastify.prefix) ?
          source.pathname.replace(fastify.prefix, rewritePrefix) : source.pathname;
        target.pathname += path.length > 0 ? ':' + path : '';
      } else {
        target = new URL(source.pathname, upstream);
      }
      target.protocol = upstream.indexOf('http:') === 0 ? 'ws:' : 'wss'
      target.search = source.search
```
Please let me know if you agreed and i commit this code snippet

Reply:
Yes, it looks more complete solution.